paddle/fluid/framework/selected_rows.h

```diff
@@ -127,9 +133,10 @@ class SelectedRows {
   // SelectedRows are simply concated when adding together. Until a
   // SelectedRows add a Tensor, will the duplicate rows be handled.
   Vector<int64_t> rows_;
+  std::unordered_map<int64_t, int64_t> id_to_index_;
```

Comment:
Please keep in mind that adding a member variable here may increase construct time of `SelectedRows`. If this is only used for dist lookuptable, can use a sub class.